### ISSUE TITLE
[FEAT] Differentiate between refetch() and normal query

### ISSUE THREAD

**Issue by KrishnaPG** [pankod/refine#2095]
**Is your feature request related to a problem? Please describe.**
We have caching behavior to reduce the latency from DB retrieval. Normal query responses are cached on the server side for short time (e.g. couple of minutes), to avoid duplicate query execution within short span of time, unless a **forced refresh** happens manually by the user where the cache will be discarded and query runs fully again.

However, on the frontend side, I do not see anyway to mark the `refetch` call as different (than the usual query of `useShow` etc.). 

In other words, the server does not know when to use cache and when to force refresh, based on the calls info.

**Describe the solution you'd like**
Can Refine mark the `refetch` calls with some kind of header (`x-refetch`) etc.? 

if there is a way for the server to differentiate between a normal query (such as `useShow`) and the `refetch` it would be great.

**Comment by KrishnaPG** [comment]
Thank you @salihozdemir 

If the `refetch` logic or the `RefreshButton` code inside Refine can pass along a custom metadata value to help differentiate the calls from normal calls, it would greatly help.

Otherwise, we have to rewire the `refetch` / `RefreshButton` logic in all places in the app code, duplicating most of the Refine's code again.

**Comment by salihozdemir** [comment]
`metaData` props can be passed to any data-related hooks and component for example `useTable`, `<DeleteButton>`, `<RefreshButton>`, useForm etc.

Since edge case is a case you should manage it with `metaData` in your `dataProvider`. But if you have a different suggestion, we'd love to hear it.

**Comment by salihozdemir** [comment]
To differentiate between refetch request and normal request, if we pass a parameter to `metaData` the client side cache of react-query will be bypassed because `metaData` is part of the query keys. We do not want this for all users.

To avoid code duplicate, you can make your own `<RefetchButton>` and even make your own CRUD components(`<Show>`, `<List>` etc.) to use in your app. If you want to change only header buttons you can use CRUD component props for passing your custom components.

**Comment by KrishnaPG** [comment]
> To differentiate between refetch request and normal request, if we pass a parameter to metaData the client side cache of react-query will be bypassed because metaData is part of the query keys. We do not want this for all users.

Then `metaData` is a wrong solution. What other ways are possible to expose the `isRefetching` property that is already exposed by the `useQuery` that Refine is not exposing to the app code?

**Comment by salihozdemir** [comment]
All of data-related hooks in refine return `queryResult`. So you can access `isRefetching` property like below:

```ts
const { queryResult } = useShow();
const { data, isLoading, isRefetching } = queryResult;
```

I used `useShow` for example purposes above. Other data-related hooks also return `queryResult`.

**Comment by omeraplak** [comment]
Hey @KrishnaPG ,
Is there anything we need to fix here?
